### PR TITLE
Improve credential filtering

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

Make the credential filtering logic more resilient by using greedier regex and adding a bunch of testing.

### Motivation

Multiple edge cases were not accounted for previously.

### CODE REVIEW

File: pkg/flare/strip.go [hunk 1]
```diff
@@ -27,43 +27,35 @@ var blankRegex = regexp.MustCompile(`^\s*$`)
...
-	dockerAPIKeyReplacer = replacer{
-		regex: regexp.MustCompile(`DD_API_KEY=['"]?\w+['"]?`),
-		replFunc: func(b []byte) []byte {
-			s := string(b)
-			s = strings.TrimSuffix(s, `"`)
-			s = strings.TrimSuffix(s, `'`)
-			replacement := "DD_API_KEY=**************************" + s[len(s)-5:]
-			return []byte(replacement)
-		},
+	apiKeyReplacer := replacer{
```
Comment (hush-hush) [inline pkg/flare/strip.go]: wouldn't this replace every string longer than 32 characters ?

File: pkg/flare/strip.go [hunk 2]
```diff
@@ -27,43 +27,35 @@ var blankRegex = regexp.MustCompile(`^\s*$`)
...
+		repl:  []byte(`://$1:********@`),
 	}
 	passwordReplacer = replacer{
-		regex: regexp.MustCompile(`( *(\w|_)*pass(word)?:).+`),
+		regex: matchYAMLKeyPart(`pass(word)?`),
 		repl:  []byte(`$1 ********`),
 	}
 	tokenReplacer = replacer{
-		regex: regexp.MustCompile(`( *(\w|_)*token:).+`),
+		regex: matchYAMLKeyPart(`token`),
```
Comment (hush-hush) [inline pkg/flare/strip.go]: Will that replace the whole line ?

Comment (arbll) [reply pkg/flare/strip.go]: It should replace everything after a YAML key containing `token` in it's name

File: pkg/flare/strip.go (hunk 1 above)
Comment (arbll) [reply pkg/flare/strip.go]: Every hexadecimal 32 character long string yes.

**Review by olivielpeau — DISMISSED** [review]
Let's add a release note, other than that LGTM (I'll let @hush-hush give the final 👍 )

File: pkg/flare/strip.go [hunk 3]
```diff
@@ -27,43 +27,35 @@ var blankRegex = regexp.MustCompile(`^\s*$`)
...
 		repl:  []byte(`$1 ********`),
 	}
 	snmpReplacer = replacer{
-		regex: regexp.MustCompile(`^(\s*community_string:) *.+$`),
+		regex: matchYAMLKey(`(community_string|authKey|privKey)`),
```
Comment (hush-hush) [inline pkg/flare/strip.go]: Why no use `matchYAMLKeyPart` ?

**Review by hush-hush — APPROVED** [review]
Tiny question, but look good to me.

**Pull request merged**